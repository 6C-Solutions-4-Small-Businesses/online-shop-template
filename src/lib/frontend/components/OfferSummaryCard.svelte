<script lang="ts">
    import Image from '$lib/frontend/components/Image.svelte'

    export let id: string
    export let name: string
    export let image: string
    export let price: number
    export let onClickHandler: () => void = () => {
    }
</script>

<button
        id="{id}"
        data-testid="{id}"
        class=" flex bg-white m-1 rounded-md border-1 border-gray-600 shadow-md hover:bg-orange-100"
        on:click={onClickHandler}
>
    <div class="w-full h-32 flex flex-row">
        <Image imageRemoteUrl="{image}" name={name} classes="w-2/5 p-2 h-full object-contain"/>
        <div class="w-3/5 flex flex-col h-w p-2 items-stretch">
            <p class="flex-1 font-thin text-xl text-left">{`${name}`}</p>
            <p class="w-full text-xl font-bold text-right">${(price / 100).toFixed(2)}</p>
        </div>
    </div>
</button>
