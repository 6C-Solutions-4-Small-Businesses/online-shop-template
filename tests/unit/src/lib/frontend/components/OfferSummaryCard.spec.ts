import {cleanup, render, type RenderResult} from '@testing-library/svelte'
import {afterEach, describe, expect} from 'vitest'
import OfferSummaryCard from '$lib/frontend/components/OfferSummaryCard.svelte'

describe('OfferSummaryCard', () => {

    const offerId = 'ordered-item-id'
    const onClick = vi.fn()

    let view: RenderResult<OfferSummaryCard, typeof import('@testing-library/dom/types/queries')>

    beforeEach(() => {

        view = render(OfferSummaryCard, {
            props: {
                id: offerId,
                name: 'ordered-item-name',
                price: 10099,
                image: 'https://cdn-cemal.nitrocdn.com/iwTMUnSsYeigzDpMWgwdBZhpcIeWwszq/assets/images/optimized/rev-59c8081/www.aaronfaber.com/wp-content/uploads/2017/03/product-placeholder-wp-95907_800x675.jpg',
                onClickHandler: onClick,
            },
        })
    })

    describe('Structure', () => {

        it('should have match snapshot', () => {

            expect(view.container).toMatchSnapshot()
        })
    })

    describe('Behavior', () => {

        it('should call onClickHandler when clicked', () => {

            const card = view.getByTestId(offerId)
            card.click()

            expect(onClick).toHaveBeenCalled()
        })
    })

    afterEach(() => {
        cleanup()
    })
})
