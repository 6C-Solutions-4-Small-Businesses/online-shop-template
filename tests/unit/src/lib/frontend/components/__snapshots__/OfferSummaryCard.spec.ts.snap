// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OfferSummaryCard > Structure > should have match snapshot 1`] = `
<body>
  <div>
    <button
      class="flex bg-white m-1 rounded-md border-1 border-gray-600 shadow-md hover:bg-orange-100"
      data-testid="ordered-item-id"
      id="ordered-item-id"
    >
      <div
        class="w-full h-32 flex flex-row"
      >
        <img
          alt="ordered-item-name"
          class="w-2/5 p-2 h-full object-contain"
          src="https://cdn-cemal.nitrocdn.com/iwTMUnSsYeigzDpMWgwdBZhpcIeWwszq/assets/images/optimized/rev-59c8081/www.aaronfaber.com/wp-content/uploads/2017/03/product-placeholder-wp-95907_800x675.jpg"
          width="160"
        />
        <!--&lt;Image&gt;-->
         
        <div
          class="w-3/5 flex flex-col h-w p-2 items-stretch"
        >
          <p
            class="flex-1 font-thin text-xl text-left"
          >
            ordered-item-name
          </p>
           
          <p
            class="w-full text-xl font-bold text-right"
          >
            $
            100.99
          </p>
        </div>
      </div>
    </button>
    <!--&lt;OfferSummaryCard&gt;-->
  </div>
</body>
`;
